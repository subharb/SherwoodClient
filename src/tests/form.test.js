import React from 'react';
import { shallow,  mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Form  from '../components/general/form';
import renderer from 'react-test-renderer';
import { FIELDS } from './helpers';

const mockStore = configureStore();
const store = mockStore();

configure({adapter: new Adapter()});

test('Form validates inputs', () => {
    
    const onSubmitFn = jest.fn(data => data);
    const formTest = mount(
        <Provider store={store}>
            <Form fields={FIELDS} callBackForm={onSubmitFn} />
        </Provider>
    );
    // let tree = formTest.toJSON();
    // expect(tree).toMatchSnapshot();
    //Doy a submit y deberá mostrar un error porque el form está vacío
    const form = formTest.find('.form');
    form.simulate('submit');
    expect(onSubmitFn).toHaveBeenCalledTimes(1);
    expect(onSubmitFn.mock.results).toEqual([{"type": "return", "value": {}}]);
    //Meto datos de prueba correctos en el form
    Object.keys(FIELDS).map(key => {
        let field = FIELDS[key];
        let input = null;
        if(field.type === "select"){
            input = formTest.find(`select[name="${key}"]`).first();
        }
        else{
            input = formTest.find(`input[name="${key}"]`).first();
        }
        input.instance().value = field.value; 
    });
    //Al dar submit es correcto
    form.simulate('submit');
    expect(onSubmitFn).toHaveBeenCalledTimes(2);
    //No me convence el resultado porque debería devolver los valores introducidos.
    expect(onSubmitFn.mock.results).toEqual([{"type": "return", "value": {}}, {"type": "return", "value": {}}]);
  });