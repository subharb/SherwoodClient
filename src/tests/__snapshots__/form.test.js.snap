// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="input-field col s12"
    >
      <label>
        <input
          checked={false}
          name="is_personal_data"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="checkbox"
          value=""
        />
        <span>
          Missing translationId: investigation.create.personal_info for language: \${ languageCode }
        </span>
      </label>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="input-field col s12"
    >
      <input
        className="validate "
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="name"
      >
        Missing translationId: investigation.create.name_field for language: \${ languageCode }
      </label>
      <span
        className="error text"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="input-field col s12"
    >
      <select
        name="type"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        value=""
      >
        <option
          disabled={true}
          value=""
        >
          Missing translationId: investigation.create.choose for language: \${ languageCode }
        </option>
        <option
          value="text"
        >
          Missing translationId: investigation.create.type_text for language: \${ languageCode }
        </option>
        <option
          value="number"
        >
          Missing translationId: investigation.create.type_number for language: \${ languageCode }
        </option>
        <option
          value="date"
        >
          Missing translationId: investigation.create.type_date for language: \${ languageCode }
        </option>
      </select>
      <label>
        Missing translationId: investigation.create.choose for language: \${ languageCode }
      </label>
      <span
        className="error text"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="input-field col s6"
    >
      <input
        className="validate "
        id="question"
        name="question"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="question"
      >
        Missing translationId: investigation.create.question_field for language: \${ languageCode }
      </label>
      <span
        className="error text"
      />
    </div>
  </div>
  <button
    className="waves-effect waves-light btn"
  >
    Missing translationId: investigation.create.add_field for language: \${ languageCode }
  </button>
</form>
`;
